externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkThriftBinaryFrontendService.scala

```diff
@@ -147,6 +147,15 @@ class SparkThriftBinaryFrontendService(
       s"${serverAddr.getHostAddress}:$portNum"
     }
   }
+
+  // When a OOM occurs, here we de-register the engine by stop its discoveryService.
+  // Then the current engine will not be connected by new client anymore but keep the existing ones
+  // alive. In this case we can reduce the engine's overhead and make it possible recover from that.
+  // We shall not tear down the whole engine by serverable.stop to make the engine unreachable for
+  // the exising clients which are still get statuses and report the end-users.
```

Comment:
1. `exising` -> `existing`
2. `report the end-users` -> `report to the end-users`